src/app/shared/courses/courses.service.ts

```diff
@@ -51,30 +58,50 @@ export class CoursesService {
             authors: null,
             isTopRated
         };
-        this.http.post('http://localhost:3004/courses/', course, httpOptions);
+        this.http.post(ROOT_URL, course, httpOptions)
+            .subscribe()
+            .add(() => { _this.preloadingService.setLoginStatus(false); });
         return course;
     }
 
-    update(id: number, name: string, length: number, description: string, isTopRated: boolean): Course {
-        const course: Course = this.getItemById(id);
-        course.name = name;
-        course.length = length;
-        course.description = description;
-        course.isTopRated = isTopRated;
-        return course;
+    update(id: number, name: string, length: number, description: string, isTopRated: boolean): void {
```

Comment:
here you need to get a whole course object which is ready for update:
update(course: Course) {
  return this.http.patch(url, course, opts);
}